domain/library/src/main/java/com/into/websoso/domain/library/model/ReadStatus.kt

```
@@ -0,0 +1,16 @@
+enum class ReadStatus(
+    val key: String,
+) {
+    WATCHING("WATCHING"),
+    WATCHED("WATCHED"),
+    QUIT("QUIT"),
+    ;
+
+    companion object {
+        fun from(key: String): ReadStatus =
+            entries.find { readStatus -> readStatus.key == key }
+                ?: throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)